### PR TITLE
Python wrapper: catalogues and global properties

### PR DESCRIPTION
**Pull request by bcumming**
Add support for mechanism catalogues, global properties, ion species specifications and so on to the Python wrapper.

### CODE REVIEW

File: python/tutorial/example1.py [hunk 1]
```diff
@@ -13,15 +13,15 @@
...
+cell.place('center', arbor.iclamp( 10, 1, 0.8))
 # Add a spike detector with threshold of -10 mV.
 cell.place('center', arbor.spike_detector(-10))
 
 # Make single cell model.
 m = arbor.single_cell_model(cell)
 
 # Attach voltage probes, sampling at 10 kHz.
-m.probe('voltage', 'center',  10000)
+m.probe('voltage', 'center',  100000)
```
Comment (thorstenhater) [inline python/tutorial/example1.py:24]: Mismatch to comment?

File: python/tutorial/example1.py [hunk 2]
```diff
@@ -13,15 +13,15 @@
 # Put hh dynamics on soma, and passive properties on the dendrites.
 cell.paint('soma', 'hh')
 # Attach stimuli with duration of 2 ms and current of 0.8 nA.
-cell.place('center', arbor.iclamp( 10, 2, 0.8))
+cell.place('center', arbor.iclamp( 10, 1, 0.8))
```
Comment (thorstenhater) [inline python/tutorial/example1.py:16]: Comment vs code

File: python/example/single_cell_swc.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import arbor
+mport arbor
```
Comment (thorstenhater) [inline python/example/single_cell_swc.py:1]: That will not work

File: arbor/morph/region.cpp [hunk 4]
```diff
@@ -416,7 +416,7 @@ struct radius_ge_ {
 };
 
 region radius_ge(region reg, double val) {
-    return region(radius_gt_{reg, val});
+    return region(radius_ge_{reg, val});
```
Comment (thorstenhater) [inline arbor/morph/region.cpp:419]: Oops ;)

**Comment by thorstenhater** [comment]
Looks good to me, except some minor issue.

File: python/tutorial/example1.py (hunk 1 above)
Comment (bcumming) [reply python/tutorial/example1.py:24]: I updated the comment.

File: python/example/single_cell_swc.py (hunk 3 above)
Comment (bcumming) [reply python/example/single_cell_swc.py:1]: It doesn't! Fixed.

File: python/tutorial/example1.py (hunk 2 above)
Comment (bcumming) [reply python/tutorial/example1.py:16]: Fixed.

**Pull request merged**